test/conftest.py

```diff
@@ -8,7 +8,7 @@
 def DockerGeneric(request, args, image, cmd, entrypoint=''):
     assert 'docker' in check_output('id'), "Are you in the docker group?"
     if 'pihole' in image:
-       args += " --dns 127.0.0.1 --dns 1.1.1.1 -v /dev/null:/etc/pihole/adlists.default -e PYTEST=1"
+       args += " --dns 127.0.0.1 --dns 1.1.1.1 -v /dev/null:/etc/pihole/adlists.default -e PYTEST=1 --cap-add=NET_ADMIN"
```

Comment:
This is what enabled part 2's fix (debug -> no-daemon) to work.

The container failed pretty badly before adding, this capability didn't seem to be required in prior versions.